test/v2/controllers/V2ArrivalsControllerSpec.scala

```diff
@@ -210,109 +218,265 @@ class V2ArrivalsControllerSpec
-          fakeRequestArrivalNotification(method = "POST", body = singleUseStringSource(<test></test>.mkString), headers = standardHeaders)
-        val result = sut.createArrivalNotification()(request)
+          fakeRequestArrivalNotification("POST", body = Source.single(ByteString(CC007C.mkString, StandardCharsets.UTF_8)), headers = standardHeadersXML)
+        val response = sut.createArrivalNotification()(request)
+
+        status(response) mustBe INTERNAL_SERVER_ERROR
+        contentAsJson(response) mustBe Json.obj(
+          "code"    -> "INTERNAL_SERVER_ERROR",
+          "message" -> "Internal server error"
+        )
+      }
+    }
+
+    "with content type set to application/json" - {
+
+      val testSinkJson: Sink[ByteString, Future[Either[FailedToValidateError, Unit]]] =
+        Flow
+          .fromFunction {
+            input: ByteString =>
+              Try(Json.parse(input.utf8String)).toEither
+                .leftMap(
+                  _ =>
+                    FailedToValidateError
+                      .JsonSchemaFailedToValidateError(NonEmptyList(JsonValidationError("path", "Invalid JSON"), Nil))
+                )
+                .flatMap {
+                  jsVal =>
+                    if ((jsVal \ "CC007").isDefined) Right(())
+                    else
+                      Left(
+                        FailedToValidateError
+                          .JsonSchemaFailedToValidateError(validationErrors = NonEmptyList(JsonValidationError("CC007", "CC007 expected but not present"), Nil))
+                      )
+                }
+          }
+          .toMat(Sink.last)(Keep.right)
+
+      val jsonValidationMockAnswer = (invocation: InvocationOnMock) =>
+        EitherT(
+          invocation
+            .getArgument[Source[ByteString, _]](1)
+            .fold(ByteString())(
+              (current, next) => current ++ next
+            )
+            .runWith(testSinkJson)
+        )
+
+      val CC007C = Json.stringify(Json.obj("CC007" -> Json.obj("test" -> "testJSON")))
+
+      val standardHeadersJSON = FakeHeaders(
+        Seq(
+          HeaderNames.ACCEPT         -> "application/vnd.hmrc.2.0+json",
+          HeaderNames.CONTENT_TYPE   -> MimeTypes.JSON,
+          HeaderNames.CONTENT_LENGTH -> contentLength
+        )
+      )
+
+      "must return Accepted when json arrival notification is considered valid" in {
+
+        when(
+          mockValidationService
+            .validateJson(eqTo(MessageType.ArrivalNotification), any[Source[ByteString, _]]())(any[HeaderCarrier], any[ExecutionContext])
+        ).thenAnswer {
+          invocation =>
+            jsonValidationMockAnswer(invocation)
+        }
+        when(mockAuditService.audit(any(), any(), eqTo(MimeTypes.JSON))(any(), any())).thenReturn(Future.successful(()))
+
+        val jsonToXmlConversion = (invocation: InvocationOnMock) =>
+          EitherT.rightT(
+            invocation.getArgument[Source[ByteString, _]](1)
```

Comment:
Rather than just returning the same thing, we should make this use a `singleUseSource` to emulate the behaviour of what is returned -- I'd imagine this would result in a test failure then for the reason I gave above.

Given this is unit testing, I don't think it matters what is returned, as we're controlling everything!